app/controllers/visualizations_controller.rb

```diff
@@ -3,20 +3,47 @@ class VisualizationsController < ApplicationController
   include HeatmapHelper
 
   def visualization
-    # TODO: (gdingle): dispatch by type
-    # type = params[:type]
-
+    # TODO: (gdingle): refactor off of heatmap
     @heatmap_data = heatmap
 
     id = params[:id]
     if id
       vis = Visualization.find(id)
+
+      # TODO: (gdingle): does this makes sense for non-heatmap?
       sample_ids = vis.data['sampleIds']
-      if sample_ids != vis.sample_ids
+      if vis.visualization_type == "heatmap" && sample_ids != vis.sample_ids
```

Comment:
Are these array guaranteed to be sorted? 
Also, again, single source of data mean you do not worry about this.

Reply:
Good question. I assumed they would be sorted because they were assigned through ActiveRecord.

Reply:
I think they normally are, but because one of them comes from the client... it will be hard to guarantee it completely.
Maybe sorting to be sure would not be a bad idea... We do not expect to have more than a few hundred samples.

Remind me again why do we keep two copies?

Reply:
I changed this to keep only a single copy in the following PR. ☺️